processing/src/main/java/io/druid/query/aggregation/cardinality/types/StringCardinalityAggregatorColumnSelectorStrategy.java

```diff
@@ -40,20 +42,28 @@ public void hashRow(DimensionSelector dimSelector, Hasher hasher)
     // nothing to add to hasher if size == 0, only handle size == 1 and size != 0 cases.
     if (size == 1) {
       final String value = dimSelector.lookupName(row.get(0));
-      hasher.putUnencodedChars(nullToSpecial(value));
+      if (NullHandling.useDefaultValuesForNull() || value != null) {
```

Comment:
Please add a comment explaining why this is written so. It's not obvious. Because different types of "replacement" are around, "default" and "special". It's not clear how do those things interact

Reply:
done.